src/main/java/org/elasticsearch/index/snapshots/IndexShardSnapshotAndRestoreService.java

```diff
@@ -80,7 +80,7 @@ public void snapshot(final SnapshotId snapshotId, final IndexShardSnapshotStatus
         }
 
         try {
-            SnapshotIndexCommit snapshotIndexCommit = indexShard.snapshotIndex();
+            SnapshotIndexCommit snapshotIndexCommit = indexShard.snapshotIndex(true);
```

Comment:
Can you add a comment why we should flush when taking a snapshot? It may be something we need to add to the documentation too.